src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisRecordCompactCtors.java

```diff
@@ -0,0 +1,17 @@
+//non-compiled with javac: Compilable with Java14
```

Comment:
@romani @pbludov should we consider creating a `javac15` CI run for input files, similar to https://github.com/checkstyle/checkstyle/blob/be3dab2a74df0abd0712df01cd93e0ad9223ee21/.ci/travis/travis.sh#L118?  Since the changes in this PR relate to jdk15 syntax, and we already have an open issue for jdk15 specific syntax (#8989), this might be a good idea.

Reply:
We we need this execution in CI

Reply:
https://github.com/checkstyle/checkstyle/issues/9007